src/main/java/org/xerial/snappy/pure/UnsafeUtil.java

```diff
@@ -66,4 +73,54 @@ public static long getAddress(Buffer buffer)
             throw new RuntimeException(e);
         }
     }
+ 
+    public static short getShort(Object inputBase, long offset)
+    {
+    	if (order.equals(ByteOrder.LITTLE_ENDIAN))
```

Comment:
Thank you for the workaround, but checking the endianness for each getXXX operation has too much overhead for a performance-sensitive library like snappy-java. Can we do this check only once and switch the implementations for little-endian and big-endian at startup time?

Reply:
I will see if I can do something

Reply:
OK, I have an alternate solution which is a bit better
I created 2 new methods SnappyRawCompressorBE and SnappyRawDecompressorBE which have the reversebytes in.
This way the LITTLE_ENDIAN code is intact.
I instead I now have IF statements in PureSnappyJava.java, to select the normal or the BE method for each Compress/Decompress method

Reply:
OK, last update. I made a new SnappyAPI interface, so there are no IF statements needed, other than on initial Loader.
So there should be no performance issues for old flows, and no net change to any of the existing modules, apart from the Loader. See 86b8578

Reply:
OK, rebased the 'net unchanged' code back to xerial base, as trivial whitespaces/tabs etc came out as changes, to make diff report come out clean.